consumers/src/unit-test/java/uk/gov/moj/sdt/consumers/ConsumerGatewayTest.java

```diff
@@ -0,0 +1,173 @@
+            assertEquals(this.soapFault.getFaultString(), "Invalid request");
+        });
+
+        assertEquals(soapFaultException.getErrorCode(), "SOAP_FAULT");
+        assertEquals(soapFaultException.getErrorDescription(), null);
+        assertEquals(soapFaultException.getCause(), null);
+    }
+
+    /**
+     * Test to verify submit query consumer does throw expected exception.
+     */
+    @Test
+    @Override
+    void testSubmitQueryRequestOutage() {
+        when(mockTransformer.transformDomainToJaxb(submitQueryRequest)).thenReturn(submitQueryRequestType);
+
+        final WebServiceException wsException = new WebServiceException();
+        wsException.initCause(new ConnectException());
+
+        when(mockClient.submitQuery(submitQueryRequestType)).thenThrow(wsException);
+
+        OutageException outageException = assertThrows(OutageException.class, () ->
+                this.consumerGateway.submitQuery(submitQueryRequest, CONNECTION_TIME_OUT, RECEIVE_TIME_OUT));
+
+        assertEquals(outageException.getErrorCode(), "OUTAGE_ERROR");
+        assertEquals(outageException.getErrorDescription(), null);
+    }
+
+    /**
+     * Test method for successful processing of submit query.
+     */
+    @Test
+    @Override
+    void testSubmitQuerySuccess() {
+        final SubmitQueryResponseType submitQueryResponseType = new SubmitQueryResponseType();
+
+        when(mockTransformer.transformDomainToJaxb(submitQueryRequest)).thenReturn(submitQueryRequestType);
+        when(mockClient.submitQuery(submitQueryRequestType)).thenReturn(submitQueryResponseType);
+        mockTransformer.transformJaxbToDomain(submitQueryResponseType, submitQueryRequest);
+
+        doAnswer (new Answer<Void>()
+        {
+            @Override
+            public Void answer(InvocationOnMock invocation) throws Throwable
+            {
+                ((SubmitQueryResponseType) invocation.getArgument(0)).setResultCount(BigInteger.valueOf(1));
+                final StatusType statusType = new StatusType ();
+                statusType.setCode (StatusCodeType.OK);
+                ((SubmitQueryResponseType) invocation.getArgument(0)).setStatus (statusType);
+                // required to be null for a void method
+                return null;
+            }
+        }).when(mockTransformer).transformJaxbToDomain(submitQueryResponseType, submitQueryRequest);
+
+        this.consumerGateway.submitQuery(submitQueryRequest, CONNECTION_TIME_OUT, RECEIVE_TIME_OUT);
+
+        verify(mockTransformer).transformDomainToJaxb(any());
+        verify(mockClient).submitQuery(any());
+
+        Assertions.assertEquals(submitQueryResponseType.getStatus().getCode().value(),
+                submitQueryRequest.getStatus(), "Status code is not equal.");
+        Assertions.assertEquals(submitQueryResponseType.getResultCount().intValue(),
+                submitQueryRequest.getResultCount(), "Result count is not equal.");
+
+    }
+
```

Comment:
Doesn't seem to be any testing of the ConsumerGateway individualRequest() method.

Reply:
There is now.